### PR TITLE
Hive mining nft

### PR DESCRIPTION
**Pull request by eonwarped**
Implements nft into mining.

Uses nft "type" property distinguish different NFTs. Mining config specifies power settings on a per type basis.

Up to four mining properties can be specified. Each of these can configure whether it is accumulated with + or * as the op.

After combining all properties of all owned NFTs, the full power is calculated by * all of the property totals.

NFTs do not have staking, so instead we use delegation to determine whether an NFT is active for mining. Delegating to the mining contract constitutes applying the nft to yourself, and delegating to others constitutes applying the nft to others.

Note that properties can be negative as well. If the op results in negative total power or NaN, note that it will result in 0 power for the account.

### CODE REVIEW

File: contracts/mining.js [hunk 1]
```diff
@@ -489,3 +718,30 @@ actions.handleStakeChange = async (payload) => {
...
+  if (api.assert(callingContractInfo && callingContractInfo.name === 'nft',
+    'must be called from nft contract')) {
+    await findAndProcessAll('mining', 'nftTokenPools', { symbol }, async (tokenPool) => {
+      const pool = await api.db.findOne('pools', { id: tokenPool.id });
+      if (pool.updating.inProgress
+          && pool.updating.tokenIndex === pool.tokenMiners.length
+          && pool.updating.lastId < nft._id) {
+        return;
+      }
+      const typeProperties = pool.nftTokenMiner.typeMap[nft.properties.type];
```
Comment (bt-cryptomancer) [inline contracts/mining.js:735]: The "type" property of an NFT instance could be changed to anything at any time.  What happens if it's not something in your typeMap when it gets delegated?

**Review by bt-cryptomancer — COMMENTED** [review]
Code mostly looks good, just have a few comments I've left at specific code lines which I'd like you to respond to before approving this.

File: contracts/mining.js (hunk 1 above)
Comment (eonwarped) [reply contracts/mining.js:734]: added type check. should be sufficient here.

**Review by bt-cryptomancer — APPROVED** [review]
changes look good

**Pull request merged**